crates/sui-framework/tests/validator_tests.move

```diff
@@ -89,34 +88,36 @@ module sui::validator_tests {
         {
             let ctx = test_scenario::ctx(scenario);
             let new_stake = coin::into_balance(coin::mint_for_testing(30, ctx));
-            validator::request_add_stake(&mut validator, new_stake, option::none(), ctx);
+            validator::request_add_delegation(&mut validator, new_stake, option::none(), sender, ctx);
 
-            assert!(validator::stake_amount(&validator) == 10, 0);
+            assert!(validator::total_stake(&validator) == 10, 0);
             assert!(validator::pending_stake_amount(&validator) == 30, 0);
         };
 
         test_scenario::next_tx(scenario, sender);
         {
-            let stake = test_scenario::take_from_sender<Stake>(scenario);
+            let delegation = test_scenario::take_from_sender<Delegation>(scenario);
+            let stake = test_scenario::take_from_sender_by_id<StakedSui>(scenario, *staking_pool::staked_sui_id(&delegation));
             let ctx = test_scenario::ctx(scenario);
-            validator::request_withdraw_stake(&mut validator, &mut stake, 5, 35, ctx);
-            test_scenario::return_to_sender(scenario, stake);
-            assert!(validator::stake_amount(&validator) == 10, 0);
+            validator::request_withdraw_delegation(&mut validator, delegation, stake, ctx);
+
+            assert!(validator::total_stake(&validator) == 10, 0);
```

Comment:
Not for this PR but in the future we can start using the nice `test_utils::assert_eq` function :D

Reply:
Yes! That would be amazing :) Happy to make a pass through at some point and convert them over :)